wgengine/router/router_linux.go

```diff
@@ -163,11 +163,15 @@ func (r *linuxRouter) Close() error {
 }
 
 // Set implements the Router interface.
-func (r *linuxRouter) Set(cfg *Config) error {
+func (r *linuxRouter) Set(cfg *Config) (err error) {
 	if cfg == nil {
 		cfg = &shutdownConfig
 	}
 
+	if err := r.dns.Set(cfg.DNS); err != nil {
```

Comment:
Maybe it should do both regardless and collect the error(s) to return later?

Reply:
Yep, sounds like a good idea. I've changed it to collect errors and then return, instead of just returning at the first error.